watchtower-plugin/src/convert.rs

```diff
@@ -258,6 +258,112 @@ impl TryFrom<serde_json::Value> for GetAppointmentParams {
     }
 }
 
+// Errors related to `getregistrationreceipt` command
+#[derive(Debug)]
+pub enum GetRegistrationReceiptError {
+    InvalidId(String),
+    InvalidFormat(String),
+}
+
+impl std::fmt::Display for GetRegistrationReceiptError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            GetRegistrationReceiptError::InvalidId(x) => write!(f, "{x}"),
+            GetRegistrationReceiptError::InvalidFormat(x) => write!(f, "{x}"),
+        }
+    }
+}
+
+// Parameters related to the `getregistrationreceipt` command
+#[derive(Debug)]
+pub struct GetRegistrationReceiptParams {
+    pub tower_id: TowerId,
+    pub subscription_start: Option<u32>,
+    pub subscription_expiry: Option<u32>,
+}
+
+impl TryFrom<serde_json::Value> for GetRegistrationReceiptParams {
+    type Error = GetRegistrationReceiptError;
+
+    fn try_from(value: serde_json::Value) -> Result<Self, Self::Error> {
+        match value {
+            serde_json::Value::Array(a) => {
+                let param_count = a.len();
+                if param_count != 1 && param_count != 3 {
```

Comment:
If you want to emphasize the boundary not being fully passed (given this may be a common error), you can have a special case here for two params before checking one or three were you error that if a boundary is set, both ends of the boundary need to be specified. e.g:

```
if parame_count == 2 {...}
else if param_count != 1 && param_count != 3 {...}
else {...}
```